ledger-api/grpc-definitions/com/daml/ledger/api/v1/command_service.proto

```diff
@@ -24,12 +24,38 @@ service CommandService {
   // Errors:
   // - ``UNAUTHENTICATED``: if the request does not include a valid access token
   // - ``PERMISSION_DENIED``: if the claims in the token are insufficient to perform a given operation
-  // - ``NOT_FOUND``: if the request does not include a valid ledger id
-  // - ``INVALID_ARGUMENT``: if the payload is malformed or is missing required fields
+  // - ``NOT_FOUND``: if the request does not include a valid ledger id or the submission rank or
+  //   deduplication period are too old.
+  // - ``INVALID_ARGUMENT`` if one of the following holds:
+  //   - The payload is malformed or is missing required fields.
+  //   - A party involved in the transaction is not known on the ledger.
+  //   - The interpreted command is inconsistent with the state of the ledger
+  //     If the submission led to a completion event, the error details' metadata key ``completion_offset``
+  //     contains the corresponding completion offset. This offset can be used as a submission rank in later calls.
+  //     If the completion event at the offset is a definite answer, the metadata key ''definite_answer``
+  //     is set to ``true``.
   // - ``RESOURCE_EXHAUSTED``: if the number of in-flight commands reached the maximum (if a limit is configured)
   // - ``UNAVAILABLE``: if the participant is not yet ready to submit commands or if the service has been shut down.
-  // - ``ABORTED``: if a contract key is missing or duplicated due to for example contention on resources
-  rpc SubmitAndWait (SubmitAndWaitRequest) returns (google.protobuf.Empty);
+  // - ``ABORTED`` if one of the following holds:
+  //   - A contract key is missing or duplicated due to for example contention on resources.
```

Comment:
```suggestion
  //   - A lock for serializability checking could not be acquired during commit time, e.g., due to contention.
```

Reply:
The contract key example seems too specific, and is also confusing wrt the the `INVALID_ARGUMENT` that I'd expect to be raised in case a `NoSuchKey` constraint is violated during commit time.